src/cfdp_core/pdu_directive.cpp

```diff
@@ -111,3 +113,119 @@ std::vector<uint8_t> cfdp::pdu::directive::AckPdu::encodeToBytes() const
 
     return encodedPdu;
 }
+
+cfdp::pdu::directive::EndOfFile::EndOfFile(Condition conditionCode, uint32_t checksum,
+                                           uint32_t fileSize)
+    : conditionCode(conditionCode), checksum(checksum), fileSize(fileSize),
+      largeFileFlag(header::LargeFileFlag::SmallFile)
+{
+    if (conditionCode != Condition::NoError)
+    {
+        throw exception::PduConstructionException(
+            "Fault location cannot be omitted with `No error` condition code");
+    }
+};
+
+cfdp::pdu::directive::EndOfFile::EndOfFile(Condition conditionCode, uint32_t checksum,
+                                           uint64_t fileSize)
+    : conditionCode(conditionCode), checksum(checksum), fileSize(fileSize),
+      largeFileFlag(header::LargeFileFlag::LargeFile)
+{
+    if (conditionCode != Condition::NoError)
+    {
+        throw exception::PduConstructionException(
+            "Fault location cannot be omitted with `No error` condition code");
+    }
+};
+
+cfdp::pdu::directive::EndOfFile::EndOfFile(Condition conditionCode, uint32_t checksum,
+                                           uint32_t fileSize, uint8_t lengthOfEntityID,
+                                           uint64_t faultEntityID)
+    : conditionCode(conditionCode), checksum(checksum), fileSize(fileSize),
+      largeFileFlag(header::LargeFileFlag::SmallFile), lengthOfEntityID(lengthOfEntityID),
+      faultEntityID(faultEntityID)
+{
+    if (conditionCode == Condition::NoError)
+    {
+        throw exception::PduConstructionException(
+            "Fault location should be omitted with `No error` condition code");
+    }
+};
+
+cfdp::pdu::directive::EndOfFile::EndOfFile(Condition conditionCode, uint32_t checksum,
+                                           uint64_t fileSize, uint8_t lengthOfEntityID,
+                                           uint64_t faultEntityID)
+    : conditionCode(conditionCode), checksum(checksum), fileSize(fileSize),
+      largeFileFlag(header::LargeFileFlag::LargeFile), lengthOfEntityID(lengthOfEntityID),
+      faultEntityID(faultEntityID)
+{
+    if (conditionCode == Condition::NoError)
+    {
+        throw exception::PduConstructionException(
+            "Fault location should be omitted with `No error` condition code");
+    }
+};
+
+cfdp::pdu::directive::EndOfFile::EndOfFile(std::span<uint8_t const> memory,
+                                           LargeFileFlag largeFileFlag, uint8_t lengthOfEntityID)
```

Comment:
how do we know in advance if incoming buffer contains a large or small file?

Reply:
header has this values, we have to decode it first

Reply:
Right